app/src/main/java/com/example/instagramclone/activities/profile/EditProfViewModel.kt

```diff
@@ -11,86 +13,60 @@ import com.google.firebase.auth.FirebaseAuth
 import com.google.firebase.auth.FirebaseUser
 import com.google.firebase.database.DatabaseReference
 import com.google.firebase.database.FirebaseDatabase
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
 
-class EditProfViewModel : ViewModel(), PasswordDialog.Listener {
+class EditProfViewModel(val editProfRepository: EditProfileRepository) : ViewModel() {
 
     val livaData = MutableLiveData<StateProfiles>()
-    var mUser: User? = null
-    var mDatabase: DatabaseReference = FirebaseDatabase.getInstance().reference
 
-    override fun onPasswordConfirm(password: String) {
-        if (!password.isNullOrEmpty()) {
-            val credential = EmailAuthProvider.getCredential(mUser!!.email, password)
-            FirebaseAuth.getInstance().currentUser!!.reauthenticate(credential){
-                updateProfile(mUser!!)
-            }
-        } else {
-            livaData.value = StateProfiles.Error("You should enter your password")
-        }
-    }
-
-    private fun updateUser(user: User) {
-        val updatesMap = mutableMapOf<String, Any>()
-        if (user.name != mUser?.name) updatesMap["name"] = user.name
-        if (user.username != mUser?.username) updatesMap["username"] = user.username
-        if (user.website != mUser?.website) updatesMap["website"] = user.website
-        if (user.bio != mUser?.bio) updatesMap["bio"] = user.bio
-        if (user.email != mUser?.email) updatesMap["email"] = user.email
-        if (user.phone != mUser?.phone) updatesMap["phone"] = user.phone
-
-
-        mDatabase.updateUser(FirebaseAuth.getInstance().currentUser!!.uid, updatesMap) {
-            livaData.value = StateProfiles.Success("Profile saved")
-
-        }
 
-    }
-
-    fun getCurrentUser() {
-        FirebaseDatabase.getInstance().reference.child("users")
-            .child(FirebaseAuth.getInstance().currentUser!!.uid)
-            .addListenerForSingleValueEvent(ValueEventListenerAdapter {
-                livaData.value = StateProfiles.CurrentUser(it.getValue(User::class.java)!!)
-                mUser = it.getValue(User::class.java)!!
+    fun onPasswordConfirm(password: String) {
+       livaData.value = StateProfiles.ShowLoading
+        CompositeDisposable().add(
+            editProfRepository.onPasswordConfirmed(password)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({
+                livaData.value = StateProfiles.SuccessPasswordConfirm
+                livaData.value = StateProfiles.HideLoading
```

Comment:
Can separate HideLoading to method .doFinally{ livaData.value = Loading.HideLoading }

Reply:
Can separate ShowLoading .doOnSubscribe{ loadingLiveData.value = Loading.Show }